students/thorn/lesson03/strformat_lab.py

```diff
@@ -5,15 +5,114 @@
 - String Formatting Lab
 """
 
+import random
+
 def task_one(in_tuple = (2, 123.4567, 10000, 12345.67)):
-    """ Takes a tuple as a parameter and formats its members as follows:
+    """ 
+    Takes a tuple as a parameter and formats its members to the following:
+    'file_002 :   123.46, 1.00e+04, 1.23e+04'
     1: Generates a filename.
     2: Float displayed with 2 decimal places.
     3: Integer displayed in scientific notation with 2 decimal places.
     4: Float displayed in scientific notation to 3 sig figs.
     """
+    new_tuple = "file_{:03d} : {:.02f}, {:.2E}, {:.2E}".format(in_tuple[0], in_tuple[1], in_tuple[2], in_tuple[3])
+    print("Task One: " + new_tuple)
+    return new_tuple
+
+def task_two(in_tuple = (2, 123.4567, 10000, 12345.67)):
+    """ 
+    Returns the same as task_one but using an alternate formatting method.
+    F strings in this case.
+    """
+    new_tuple = f"file_{in_tuple[0]:03d} : {in_tuple[1]:.02f}, {in_tuple[2]:.2E}, {in_tuple[3]:.2E}"
+    print("Task Two: " + new_tuple)
+    return new_tuple
+
+def task_three(in_tuple):
+    """
+    Rewrites "the 3 numbers are: {:d}, {:d}, {:d}".format(1,2,3) to take an
+    arbitrary amount of values.
+    """
+    in_tuple_length = len(in_tuple)
+    # Create format string.
+    base_add = "{:d}, "
+    base_string = f"The {in_tuple_length} numbers are: {base_add * in_tuple_length}"
+    print(f"Task 3: {base_string.format(*in_tuple)}")
+    return base_string.format(*in_tuple)
+    
+def task_four(in_tuple = (4, 30, 2017, 2, 27)):
+    """ 
+    Takes a tuple and [0][1][2][3][4] and returns [3][4][2][0][1] with a 0
+    padding for single digit numbers.
+    Final output: '02 27 2017 04 30'
+    """
+    new_tuple = "{3:02d} {4} {2} {0:02d} {1}".format(*in_tuple)
+    print("Task 4: " + new_tuple)
+    return new_tuple
+
+def task_five(in_list = ['oranges', 1.3, 'lemons', 1.1]):
+    """ 
+    Takes a four element list ['oranges', 1.3, 'lemons', 1.1] and displays
+    "The weight of an orange is 1.3 and the weight of a lemon is 1.1" with
+    f-strings.
+    It then displays the names of the fruit in upper case, and the weight 20% 
+    higher.
+    """
+    # Eliminate the end 's'.
+    new_list = []
+    for item in in_list:
+        try:
+            if item[-1].lower() == 's':
+                new_item = item[:-1]
+                new_list.append(new_item)
+        except TypeError:
+            new_list.append(item)
+    first_string = f"The weight of an {new_list[0]} is {new_list[1]} and the weight of a {new_list[2]} is {new_list[3]}."
```

Comment:
you could also remove the 's' inline -> `f"an {in_list[:-1]}"`



the for loop is useful though for a dynamic type of solution